src/main/java/net/minecraftforge/client/model/ModelFluid.java

```diff
@@ -272,12 +308,14 @@ public BakedFluid(Optional<TRSRTransformation> transformation, ImmutableMap<Tran
                 {
                     side = EnumFacing.getHorizontal((5 - i) % 4);
                     BakedQuad q[] = new BakedQuad[2];
+                    boolean so = overlay.isPresent() && sideOverlays[side.getHorizontalIndex()];
+                    TextureAtlasSprite texture = so ? overlay.get() : flowing;
 
-                    for(int k = 0; k < 2; k++)
+                    for (int k = (so ? 1 : 0); k < 2; k++)
```

Comment:
I can't easily understand what is happening on this line.
The style of this file is a bit too terse I think, can you change some of these variable names to be more descriptive or add some comments?